src/scales/scale.time.js

```diff
@@ -395,10 +395,17 @@ function getLabelTimestamps(scale) {
 
 function getAllTimestamps(scale) {
 	var timestamps = scale._cache.all || [];
+	var label, data;
 
 	if (!timestamps.length) {
-		timestamps = getDataTimestamps(scale).concat(getLabelTimestamps(scale));
-		timestamps = scale._cache.all = arrayUnique(timestamps).sort(sorter);
+		data = getDataTimestamps(scale);
+		label = getLabelTimestamps(scale);
+		if (data.length && label.length) {
+			timestamps = arrayUnique(data.concat(label)).sort(sorter);
```

Comment:
We could end up doing `arrayUnique` and `sort` three times. Once in `getDataTimestamps, once in `getLabelTimestamps`, and once here. I think it'd be preferable if we could make it work more like it did before the early parse PR where it could happen only once

Reply:
I wonder why the `uPlot` benchmark was done with `distribution: 'series'`
If time is plotted linearly (default), we can get away without any `arrayUnique` calls.

Reply:
probably because your timeseries demo uses 'series' :D 

https://www.chartjs.org/samples/latest/scales/time/financial.html

Reply:
Seems it is done for demonstrating weekend exclusion from the chart.

Reply:
interesting. something for me to think about for https://leeoniya.github.io/uPlot/demos/high-low-bands.html

Reply:
We should also be able to get rid of `arrayUnique` if we allow passing in the data as a single dataset instead of making the user specify the timestamps again for each line https://github.com/chartjs/Chart.js/issues/6696

Reply:
That can already be done by using `labels` and (primitive) `data`, and `ticks.source='labels'`.
Should try out the new data format(s), how it would work out.